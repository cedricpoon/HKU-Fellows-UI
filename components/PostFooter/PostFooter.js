import React, { Component } from 'react';
import { Share } from 'react-native';
import PropTypes from 'prop-types';
import { Footer, FooterTab, Button, Icon } from 'native-base';

import { deepLink } from 'hkufui/config';
import appString from 'hkufui/app.json';

class PostFooter extends Component {
  async _share() {
    const { sharePayload, title, subtitle } = this.props;
    const _subtitle = subtitle ? ` Â· ${subtitle}` : '';
    try {
      await Share.share({
        message: `[${appString.displayName}]\n${title}${_subtitle}\n${deepLink.prefix}${deepLink.post(sharePayload)}`
      })
    } catch (error) {
      return;
    }
  }

  render() {
    const { firstPage, lastPage, onPageChangeThunk, onRefresh, enableRefresh, onReply } = this.props;

    return (
      <Footer>
        <FooterTab>
          <Button transparent={firstPage} disabled={firstPage} onPress={onPageChangeThunk(-1)}>
            <Icon name="arrow-left" type="MaterialCommunityIcons" />
          </Button>
          <Button onPress={onRefresh} transparent={!enableRefresh} disabled={!enableRefresh} >
            <Icon name="refresh" type="MaterialCommunityIcons" />
          </Button>
          <Button onPress={onReply} transparent={!enableRefresh} disabled={!enableRefresh}>
            <Icon name="comment-text-outline" type="MaterialCommunityIcons" />
          </Button>
          <Button onPress={this._share.bind(this)}>
            <Icon name="share-variant" type="MaterialCommunityIcons" />
          </Button>
          <Button transparent={lastPage} disabled={lastPage} onPress={onPageChangeThunk(1)}>
            <Icon name="arrow-right" type="MaterialCommunityIcons" />
          </Button>
        </FooterTab>
      </Footer>
    );
  }
}

PostFooter.defaultProps = {
  onPageChangeThunk: () => { return null },
  enableRefresh: true
}

PostFooter.propTypes = {
  firstPage: PropTypes.bool,
  lastPage: PropTypes.bool,
  onPageChangeThunk: PropTypes.func,
  onRefresh: PropTypes.func,
  onReply: PropTypes.func,
  enableRefresh: PropTypes.bool,
  sharePayload: PropTypes.string,
  title: PropTypes.string,
  subtitle: PropTypes.string
}

export default PostFooter;
