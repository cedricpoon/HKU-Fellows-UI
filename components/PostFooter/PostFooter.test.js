import React from 'react';
import { shallow } from 'enzyme';
import PostFooter from './PostFooter';

describe('Testing PostFooter Component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <PostFooter />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('is in first page', () => {
    const wrapper = shallow(
      <PostFooter firstPage />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('is in last page', () => {
    const wrapper = shallow(
      <PostFooter lastPage />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('contains page change callback', () => {
    const wrapper = shallow(
      <PostFooter onPageChangeThunk={() => { return () => {} }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
