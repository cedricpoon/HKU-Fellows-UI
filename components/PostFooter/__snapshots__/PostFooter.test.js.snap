// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PostFooter Component is in first page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostFooter
    firstPage={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(FooterTab)>
        <Styled(Button)
          disabled={true}
          transparent={true}
        >
          <Styled(Icon)
            name="arrow-left"
            type="MaterialCommunityIcons"
          />
        </Styled(Button)>
        <Styled(Button)>
          <Styled(Icon)
            name="share-variant"
            type="MaterialCommunityIcons"
          />
        </Styled(Button)>
        <Styled(Button)>
          <Styled(Icon)
            name="comment-text"
            type="MaterialCommunityIcons"
          />
        </Styled(Button)>
        <Styled(Button)>
          <Styled(Icon)
            name="arrow-right"
            type="MaterialCommunityIcons"
          />
        </Styled(Button)>
      </Styled(FooterTab)>,
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Button)
            disabled={true}
            transparent={true}
          >
            <Styled(Icon)
              name="arrow-left"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>,
          <Styled(Button)>
            <Styled(Icon)
              name="share-variant"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>,
          <Styled(Button)>
            <Styled(Icon)
              name="comment-text"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>,
          <Styled(Button)>
            <Styled(Icon)
              name="arrow-right"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Icon)
              name="arrow-left"
              type="MaterialCommunityIcons"
            />,
            "disabled": true,
            "transparent": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "arrow-left",
              "type": "MaterialCommunityIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Icon)
              name="share-variant"
              type="MaterialCommunityIcons"
            />,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "share-variant",
              "type": "MaterialCommunityIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Icon)
              name="comment-text"
              type="MaterialCommunityIcons"
            />,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "comment-text",
              "type": "MaterialCommunityIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Icon)
              name="arrow-right"
              type="MaterialCommunityIcons"
            />,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "arrow-right",
              "type": "MaterialCommunityIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(FooterTab)>
          <Styled(Button)
            disabled={true}
            transparent={true}
          >
            <Styled(Icon)
              name="arrow-left"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>
          <Styled(Button)>
            <Styled(Icon)
              name="share-variant"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>
          <Styled(Button)>
            <Styled(Icon)
              name="comment-text"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>
          <Styled(Button)>
            <Styled(Icon)
              name="arrow-right"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>
        </Styled(FooterTab)>,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Button)
              disabled={true}
              transparent={true}
            >
              <Styled(Icon)
                name="arrow-left"
                type="MaterialCommunityIcons"
              />
            </Styled(Button)>,
            <Styled(Button)>
              <Styled(Icon)
                name="share-variant"
                type="MaterialCommunityIcons"
              />
            </Styled(Button)>,
            <Styled(Button)>
              <Styled(Icon)
                name="comment-text"
                type="MaterialCommunityIcons"
              />
            </Styled(Button)>,
            <Styled(Button)>
              <Styled(Icon)
                name="arrow-right"
                type="MaterialCommunityIcons"
              />
            </Styled(Button)>,
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="arrow-left"
                type="MaterialCommunityIcons"
              />,
              "disabled": true,
              "transparent": true,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "arrow-left",
                "type": "MaterialCommunityIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="share-variant"
                type="MaterialCommunityIcons"
              />,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "share-variant",
                "type": "MaterialCommunityIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="comment-text"
                type="MaterialCommunityIcons"
              />,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "comment-text",
                "type": "MaterialCommunityIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="arrow-right"
                type="MaterialCommunityIcons"
              />,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "arrow-right",
                "type": "MaterialCommunityIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing PostFooter Component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostFooter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(FooterTab)>
        <Styled(Button)>
          <Styled(Icon)
            name="arrow-left"
            type="MaterialCommunityIcons"
          />
        </Styled(Button)>
        <Styled(Button)>
          <Styled(Icon)
            name="share-variant"
            type="MaterialCommunityIcons"
          />
        </Styled(Button)>
        <Styled(Button)>
          <Styled(Icon)
            name="comment-text"
            type="MaterialCommunityIcons"
          />
        </Styled(Button)>
        <Styled(Button)>
          <Styled(Icon)
            name="arrow-right"
            type="MaterialCommunityIcons"
          />
        </Styled(Button)>
      </Styled(FooterTab)>,
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Button)>
            <Styled(Icon)
              name="arrow-left"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>,
          <Styled(Button)>
            <Styled(Icon)
              name="share-variant"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>,
          <Styled(Button)>
            <Styled(Icon)
              name="comment-text"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>,
          <Styled(Button)>
            <Styled(Icon)
              name="arrow-right"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Icon)
              name="arrow-left"
              type="MaterialCommunityIcons"
            />,
            "disabled": undefined,
            "transparent": undefined,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "arrow-left",
              "type": "MaterialCommunityIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Icon)
              name="share-variant"
              type="MaterialCommunityIcons"
            />,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "share-variant",
              "type": "MaterialCommunityIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Icon)
              name="comment-text"
              type="MaterialCommunityIcons"
            />,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "comment-text",
              "type": "MaterialCommunityIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Icon)
              name="arrow-right"
              type="MaterialCommunityIcons"
            />,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "arrow-right",
              "type": "MaterialCommunityIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(FooterTab)>
          <Styled(Button)>
            <Styled(Icon)
              name="arrow-left"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>
          <Styled(Button)>
            <Styled(Icon)
              name="share-variant"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>
          <Styled(Button)>
            <Styled(Icon)
              name="comment-text"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>
          <Styled(Button)>
            <Styled(Icon)
              name="arrow-right"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>
        </Styled(FooterTab)>,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Button)>
              <Styled(Icon)
                name="arrow-left"
                type="MaterialCommunityIcons"
              />
            </Styled(Button)>,
            <Styled(Button)>
              <Styled(Icon)
                name="share-variant"
                type="MaterialCommunityIcons"
              />
            </Styled(Button)>,
            <Styled(Button)>
              <Styled(Icon)
                name="comment-text"
                type="MaterialCommunityIcons"
              />
            </Styled(Button)>,
            <Styled(Button)>
              <Styled(Icon)
                name="arrow-right"
                type="MaterialCommunityIcons"
              />
            </Styled(Button)>,
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="arrow-left"
                type="MaterialCommunityIcons"
              />,
              "disabled": undefined,
              "transparent": undefined,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "arrow-left",
                "type": "MaterialCommunityIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="share-variant"
                type="MaterialCommunityIcons"
              />,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "share-variant",
                "type": "MaterialCommunityIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="comment-text"
                type="MaterialCommunityIcons"
              />,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "comment-text",
                "type": "MaterialCommunityIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="arrow-right"
                type="MaterialCommunityIcons"
              />,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "arrow-right",
                "type": "MaterialCommunityIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
